query_select.go

```diff
@@ -923,6 +923,17 @@ func (q *SelectQuery) Count(ctx context.Context) (int, error) {
 		return 0, q.err
 	}
 
+	if err := q.forEachInlineRelJoin(func(j *relationJoin) error {
+		j.applyTo(q)
```

Comment:
It looks like this function modifies the `q`. Meaning that it will be applied twice when run with `ScanAndCount`:

- first it will be applied by Scan
- second it will be applied by Count

If that is true, then the query generation becomes racy and unsafe....

Reply:
ScanAndCount calls the unexported `count` function that corresponds to `Count` on the main branch. If I remember correctly the lines you mention do in fact modify `q` but they are only run when `Count` (not `count`) is called directly.  

I made this PR a while ago so I might be missing something, you tell me